### PR TITLE
clarify the comparison between 'our' and 'use vars'

### PR DESCRIPTION
**Pull request by book**
Fixes GH #22424

This set of changes does not require a perldelta entry.

### CODE REVIEW

File: pod/perlfunc.pod [hunk 1]
```diff
@@ -5377,9 +5377,9 @@ values:
 
     our ( undef, $min, $hour ) = localtime;
 
-L<C<our>|/our VARLIST> differs from L<C<use vars>|vars>, which allows
-use of an unqualified name I<only> within the affected package, but
-across scopes.
+L<C<our>|/our VARLIST> is lexically scoped, and as such differs from
```
Comment (guest20) [inline pod/perlfunc.pod]: Won't `lexical` here raise more questions than it answers? `my` is for lexically scoped variables.

Comment (demerphq) [reply pod/perlfunc.pod]: The problem is that with global variables "our" is lexcially scoped for the "handle" to the variable, not the variable itself. Global variables are created when used (in any way) and live in a global nested hash table (called the stash), which is entirely unrelated to whether you use "our" or "use vars" or use fully qualified notation, or if you use glob notation.

This means it is hard to talk about as the concepts do not map nicely to normal variables declarations in other languages (although there is some commonality with Javascript 'var' variables). Consider the following code.

```
package Whatever;
sub foo {};
{
   our $foo; # makes *Whatever::foo{SCALAR} available in the current scope under the unqualified name $foo.
}
# *Whatever::foo{SCALAR} still exists here, it is just that the var name $foo no longer aliases it.
{
   our $foo; # and we can access *Whatever::foo{SCALAR} here again, its the same var as the previous our
}
```

In this code we see two "our $foo" declarations, neither "creates" the variable $foo, (although they both ensure that it has been created), in fact it is created by the 'sub foo{}' statement when the Whatever::foo glob is created.

This is entirely unlike true lexical variables. Personally i prefer 'use vars' over 'our' I think the semantics map to what happens under the hood better.

Comment (Leont) [reply pod/perlfunc.pod]: Something like "it creates a lexically scoped alias"?

Comment (demerphq) [reply pod/perlfunc.pod]: @leont that would make sense yes.

Comment (mauke) [reply pod/perlfunc.pod]: > This means it is hard to talk about as the concepts do not map nicely to normal variables declarations in other languages

Perl `our` is pretty much C `extern`:
```c
void foo(void) {
    {
        extern int bar;  // global variable, must be defined elsewhere
        printf("%d\n", bar);
    }
    bar = 0;  // error, bar is not in scope here
}
```

**Comment by book** [comment]
I've updated the wording following @Leont's suggestion.

File: pod/perlfunc.pod (hunk 1 above)
Comment (leonerd) [reply pod/perlfunc.pod]: How about:

```pod
L<C<our>|/our VARLIST> creates a lexically scoped alias to the package variable of the same name, and as such differs from ...
```

Thus pointing out that it is the *alias* that is lexically scoped, to the variable that is package scoped.

**Comment by book** [comment]
And now updated the wording following @leonerd's suggestion.

**Review by leonerd — APPROVED** [review]
LGTM

**Pull request merged**